beauty-consulting-app/app/Pages/ProfileView.js

```diff
@@ -1,119 +1,437 @@
+			concerns: updatedConcerns,
+		};
+
+		try {
+			const apiURL =
+				Constants.manifest?.extra?.apiUrl ||
+				process.env.EXPO_PUBLIC_API_URL;
+			if (!apiURL) {
+				console.error("apiURL not defined");
+				Alert.alert("Error", "API URL is not defined.");
+				setLoading(false);
+				return;
 			}
+			const res = await axios.put(
+				`${apiURL}/client/${userContext.username}`,
+				req
+			);
+			console.log("update successful: ", res.data);
 
-			// update user context for rest of session
-			userContext = userContext.updateUserContext({
+			// Update UserContext with the new information
+			userContext.updateUserContext({
 				username: userContext.username,
-				name: name,
-				age: age,
-				gender: gender,
+				name: updatedName,
+				age: userContext.age,
+				gender: updatedGender,
 				phoneNumber: userContext.phoneNumber,
 				email: userContext.email,
 				hairDetails: userContext.hairDetails,
-				allergies: allergies,
-				concerns: concerns,
-				updateUserContext: userContext.updateUserContext,
+				allergies: updatedAllergies,
+				concerns: updatedConcerns,
 			});
+
+			Alert.alert("Success", "Profile updated successfully.");
+		} catch (error) {
+			console.error("Error with request: ", error);
+			Alert.alert(
+				"Error",
+				"Failed to update profile. Please try again later."
+			);
+		} finally {
+			setLoading(false);
 		}
-		setIsEdit(!isEdit);
 	};
 
+	// Function to handle account deletion
 	const deleteAccount = async () => {
 		// TODO: update with a separate flow for password validation after backend rework
 
+		setLoading(true);
+
 		try {
 			console.log(userContext);
-			const res = api.delete(`/client/${userContext.username}`);
+			const apiURL =
+				Constants.manifest?.extra?.apiUrl ||
+				process.env.EXPO_PUBLIC_API_URL;
+			if (!apiURL) {
+				console.error("apiURL not defined");
+				Alert.alert("Error", "API URL is not defined.");
+				setLoading(false);
+				return;
+			}
+			const res = await axios.delete(
+				`${apiURL}/client/${userContext.username}`
+			);
 			console.log(res.message);
+			Alert.alert(
+				"Account Deleted",
+				"Your account has been successfully deleted.",
+				[
+					{
+						text: "OK",
+						onPress: () => navigation.navigate("Sign In"),
+					},
+				]
+			);
 		} catch (error) {
 			console.error("Error with request: ", error);
-			return;
+			Alert.alert(
+				"Error",
+				"Failed to delete account. Please try again later."
+			);
+		} finally {
+			setLoading(false);
+			setModalVisible(false);
 		}
-		navigation.navigate("Sign In");
 	};
 
+	// Function to fetch user data on component mount (optional)
+	useEffect(() => {
+		const fetchUserData = async () => {
+			setLoading(true);
+			try {
+				const apiURL =
+					Constants.manifest?.extra?.apiUrl ||
+					process.env.EXPO_PUBLIC_API_URL;
+				if (!apiURL) {
+					console.error("apiURL not defined");
+					Alert.alert("Error", "API URL is not defined.");
+					setLoading(false);
+					return;
+				}
```

Comment:
This should not be necessary. See the refactored way to make http calls to the backend in #89.

Reply:
Resolved.

Reply:
This is still not quite right. All you should need is just a call to `api.get`

Reply:
Also I am still unable to see the profile photo after logging out. It seems like the photo is being stored properly in the database though.